rpc.if

```diff
@@ -132,6 +132,43 @@ interface(`rpc_write_exports',`
 	allow $1 exports_t:file write_file_perms;
 ')
 
+########################################
+## <summary>
+##	Allow adding/removing exports files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`rpc_add_rm_exports',`
+	gen_require(`
+		type exports_t;
+	')
+
+	allow $1 exports_t:file { create unlink };
+')
+
+
+########################################
+## <summary>
+##	Allow export directory operations
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`rpc_dir_exports',`
+	gen_require(`
+		type exports_t;
+	')
+
+	allow $1 exports_t:dir { search read open write getattr add_name remove_name };
```

Comment:
Instead of enumerating permissions, use `rw_dir_perms` pattern, or even `manage_files_pattern()` interface which could also cover the files handling.

Reply:
Removed the two interfaces I created in rpc and replaced with a single `rpc_manage_exports` which uses the `manage_files_pattern`